### ISSUE TITLE
using protoc 3.3.0 generate wrong parseMethod？

### ISSUE THREAD

**Issue by SeriousMa** [grpc/grpc-java#3148]
Please answer these questions before submitting your issue.

### What version of gRPC are you using?
grpc 1.4.0
using protoc generate version 3.2.0
![image](https://user-images.githubusercontent.com/464213/27628792-b5e28a66-5c22-11e7-9be8-491727a59127.png)

using protoc generate version 3.3.0
![image](https://user-images.githubusercontent.com/464213/27628833-d0f3bce4-5c22-11e7-8015-dc661c16b6a7.png)

**Comment by ejona86** [comment]
Do you get an error? I'd expect to see an override of the method. For example, the checked in sources in grpc-java (built with protoc 3.3.0) have the [`ByteBuffer` version](https://github.com/grpc/grpc-java/blob/v1.4.0/testing-proto/src/generated/main/java/com/google/protobuf/EmptyProtos.java#L148) but they still have the [`ByteString` version](https://github.com/grpc/grpc-java/blob/v1.4.0/testing-proto/src/generated/main/java/com/google/protobuf/EmptyProtos.java#L159).

**Comment by dapengzhang0** [comment]
seems I also hit this issue in the second commit of #3151

**Comment by SeriousMa** [comment]
same way #3149 
 i exclusion protobuf-lite and it's ok now
```
 <dependency>
                <groupId>io.grpc</groupId>
                <artifactId>grpc-all</artifactId>
                <version>${grpc.version}</version>
                <exclusions>
                    <exclusion>
                        <artifactId>protobuf-lite</artifactId>
                        <groupId>com.google.protobuf</groupId>
                    </exclusion>
                </exclusions>
            </dependency>
```

**Comment by ejona86** [comment]
Fixed by #3153. Manually excluding protobuf-lite is a workaround for now.
